Library/Homebrew/dev-cmd/contributions.rb

```diff
@@ -0,0 +1,100 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  extend T::Sig
+
+  module_function
+
+  SUPPORTED_REPOS = %w[brew core cask bundle].freeze
+
+  sig { returns(CLI::Parser) }
+  def contributions_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `contributions [email]`
+
+        Contributions to Homebrew repos for a user.
+      EOS
+
+      flag "--from=",
+           description: "Date (ISO-8601 format) to start searching contributions."
+
+      flag "--to=",
+           description: "Date (ISO-8601 format) to stop searching contributions."
+
+      comma_array "--repos=",
+                  description: "The Homebrew repositories to search for contributions in. " \
+                               "Comma separated. Supported repos: #{SUPPORTED_REPOS.join(", ")}."
+
+      named_args :email
+    end
+  end
+
+  sig { returns(NilClass) }
+  def contributions
+    args = contributions_args.parse
+
+    return ofail "`--repos` is required." if args[:repos].empty?
```

Comment:
Usually we use methods to get arguments (i.e. `args.repos`, `args.from`, `args.to`). TBH I didn't even realize you could do this. I think for consistency using the methods may be better, but I guess if it works I'm not opposed

```suggestion
    return ofail "`--repos` is required." if args.repos.empty?
```

Reply:
Also iirc there's a way to tell the argument parser this argument is required i.e. by requiring multiple named args like `email repo1 [repo2]` etc. That feels more in keeping with how we require arguments (i.e. never requiring flags)

Reply:
> Usually we use methods to get arguments (i.e. args.repos, args.from, args.to).

I wanted to do this, but Sorbet is very unhappy when I do (I guess that's why we don't typecheck a lot of our `dev-cmd/` commands, huh):

```
dev-cmd/contributions.rb:94: Method to does not exist on Homebrew::CLI::Args https://srb.help/7003
    94 |    cmd << "--before=#{args.to}" if args.to
                                    ^^
  Got Homebrew::CLI::Args originating from:
    dev-cmd/contributions.rb:92:
    92 |  def git_log_author_cmd(repo_path, args)
```

Reply:
If you add the methods to [`args.rbi`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cli/args.rbi) this won't be an issue. That's what we do elsewhere

Reply:
Done, and I'm pretty sure I've fixed the argument parsing so it accepts `repos` as a second, required, named, argument, so usage is now like `brew contributions me@issyl0.co.uk all` or `brew contributions me@issyl0.co.uk brew,core,cask`.